framework/src/vectorpostprocessors/ConstantVectorPostprocessor.C

```diff
@@ -43,10 +43,19 @@ ConstantVectorPostprocessor::ConstantVectorPostprocessor(const InputParameters &
   for (unsigned int j = 0; j < nvec; ++j)
     _value[j] = &declareVector(names[j]);
 
-  std::vector<std::vector<Real>> v = getParam<std::vector<std::vector<Real>>>("value");
-  if (v.size() != nvec)
-    paramError("value",
-               "Leading dimension must be equal to leading dimension of vector_names parameter.");
+  std::vector<std::vector<Real>> v;
+  if (isParamValid("value"))
+  {
+    v = getParam<std::vector<std::vector<Real>>>("value");
+    if (v.size() != nvec)
+      paramError("value",
+                 "Leading dimension must be equal to leading dimension of vector_names parameter.");
+  }
+  else
+  {
+    for (unsigned int i = 0; i < nvec; ++i)
+      v.push_back({std::numeric_limits<Real>::max()});
```

Comment:
Not sure this is needed. You probably just keep the vectors empty if `value` is not specified.

Reply:
since it's a vector of vectors I need to initialize it with a vector of size 1 so that I can transfer the rest of the vector into it later.   I filled it with numeric limits so that it would be obviously wrong if it didn't get filled or if it was used before it was transferred into.

Reply:
This could resize a bunch of times.
This might work as a replacement to the for loop: v.resize(nvec, {std::numeric_limits<Real>::max()});